lib/build-utilities/migrate-initializers.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const path = require('path');
+const existsSync = require('exists-sync');
+const fs = require('fs-extra');
+
+const fastbootInitializerTypes = [ 'initializers', 'instance-initializers'];
+const FASTBOOT_DIR = 'fastboot';
+
+/**
+ * Helper function to check if there are any `(instance-)?intializers/[browser|fastboot]/` path under the
+ * given root path.
+ *
+ * @param {String} rootPath
+ * @param {String} type
+ * @returns {Boolean} true if path exists
+ */
+function _checkInitializerTypeExists(rootPath, type) {
+  const isTypeExists = fastbootInitializerTypes.some((fastbootInitializerType) => {
+    const pathType = path.join(rootPath, 'app', fastbootInitializerType, type);
+
+    return existsSync(pathType);
+  });
+
+  return isTypeExists;
+}
+/**
+ * Helper function to check if there are any `(instance-)?intializers/browser/` path under the
+ * given root path and throw an error
+ *
+ * @param {String} rootPath
+ */
+function _checkBrowserInitializers(rootPath) {
+  const isBrowserInitializersPresent = _checkInitializerTypeExists(rootPath, 'browser');
+
+  if (isBrowserInitializersPresent) {
+    var errorMsg = `FastBoot build no longer supports ${rootPath}/app/(instance-)?initializers/browser structure. ` +
+                         `Please refer to www.ember-fastboot.com for a migration path.`;
```

Comment:
Yeah we can't auto-migrate browser based initializers, hence we will need to fail the build. I am happy to refactor the error messages to be more better. Probably sometime this week i'll come up with a migration cheat sheet so that it is easier for addons 😄